src/lib.rs

```diff
@@ -694,11 +694,50 @@ impl Log for NopLogger {
 
 /// The location of a log message.
 ///
+/// # Use
+///
+/// `Location` structs are created by the [`log!`] macro. They are attached to
+/// [`Record`] structs, which are used by loggers to display logging messages.
+/// `Location`s can be accessed using the [`location()`] method on [`Record`]s.
+/// Log users do not need to directly use this struct.
+///
+/// ## Example
+/// The below example shows a simple logger that prints the module path,
+/// file name, and line number of the location the [`log!`] macro was called.
+///
+/// ```rust
+/// # extern crate log;
+/// struct SimpleLogger;
+///
+/// impl log::Log for SimpleLogger {
+///     fn enabled(&self, metadata: &log::Metadata) -> bool {
+///         true
+///     }
+///
+///     fn log(&self, record: &log::Record) {
+///         if !self.enabled(record.metadata()) {
+///             return;
+///         }
+///
+///         let location = record.location();
+///         println!("{}:{}:{} -- {}",
+///                  location.module_path(),
+///                  location.file(),
+///                  location.line(),
+///                  record.args());
+///     }
+/// }
+/// ```
+///
 /// # Warning
 ///
 /// The fields of this struct are public so that they may be initialized by the
-/// `log!` macro. They are subject to change at any time and should never be
+/// [`log!`] macro. They are subject to change at any time and should never be
 /// accessed directly.
+///
+/// [`log!`]: macro.log!.html
```

Comment:
`macro.log!.html` redirects to `macro.log.html` so you would save the user a redirect by pointing to `macro.log.html` directly.